internal/security/redis/main.go

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright 2020 Redis Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Andre Srinivasan
+ *******************************************************************************/
+
+package redis
+
+import (
+	"context"
+	"os"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/handlers/database"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/telemetry"
+	"github.com/edgexfoundry/edgex-go/internal/security/redis/config"
+	"github.com/edgexfoundry/edgex-go/internal/security/redis/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/flags"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/handlers/httpserver"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/handlers/secret"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/interfaces"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/startup"
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/gorilla/mux"
+)
+
+func Main(ctx context.Context, cancel context.CancelFunc, router *mux.Router, readyStream chan<- bool) {
+	startupTimer := startup.NewStartUpTimer(clients.SecurityBootstrapRedisKey)
+
+	// All common command-line flags have been moved to DefaultCommonFlags.
+	f := flags.New()
+	f.Parse(os.Args[1:])
+
+	configuration := &config.ConfigurationStruct{}
+	dic := di.NewContainer(di.ServiceConstructorMap{
+		container.ConfigurationName: func(get di.Get) interface{} {
+			return configuration
+		},
+	})
+
+	httpServer := httpserver.NewBootstrap(router, false) // Not a real HTTP server; false == don't listen/serve
+
+	// bootstrap.Run is needed for the underlying configuration system. Conveniently, it also creates
+	// a pipeline of functions as the list of BootstrapHandler's is executed in order. Note that
+	// bootstrap.Run will return when the last thread spawned by the the BoostrapHandler's returns. In
+	// this case, telemetry.BootstrapHandler creates a thread that effectively will never return
```

Comment:
I don't like this. This is a hack to keep the service running which does take CPU cycles. The service should just go to sleep at the end. Looks like this select will sleep forever and exit on SIGTERM.

```
   select {
   case <-ctx.Done():
      lc.Info(<your exiting message>)
      os.exit(0)
   }
``` 

I think you should use bootstrap.RunAndReturnWaitGroup(...) and have the code above after it returns.

Reply:
Done

Reply:
So now I'm confused, I thought this was supposed to be a one-shot service which reads the creds from vault, and then sets them in redis on startup. Is there a reason it should be long-running?

Reply:
Hmmm, that sounds a bit ugly, as we have other services that are one-shot (e.g. security-*-setup)...

@andresrinivasan is this a one-time configuration of the password, or does this happen every time the EdgeX instance is brought up?

Reply:
> that sounds a bit ugly,

@tonyespy , Yes from a Snaps perspective it is ugly... 
@andresrinivasan, @bnevis-i ,  I think the service should not be `long` running and just exit when it is finished. For docker we can have an entry point script that handles the go to sleep rather than stop container. See this one:
https://github.com/edgexfoundry/edgex-go/blob/master/cmd/security-secretstore-setup/entrypoint.sh#L65

Also I just noticed that that entry point script is referencing the old `security-secretstore-read` so that script needs to be reworked as part of this PR.

Reply:
I will update the service code to exit and add the entry point script to the container.